src/valkey.c

```diff
@@ -854,6 +854,10 @@ valkeyContext *valkeyConnectWithOptions(const valkeyOptions *options) {
         c->flags |= VALKEY_PREFER_IPV6;
     }
 
+    if (options->options & VALKEY_OPT_MPTCP) {
+        c->flags |= VALKEY_MPTCP;
+    }
```

Comment:
Shall we set the error here?

```suggestion
    if (options->options & VALKEY_OPT_MPTCP) {
#ifndef IPPROTO_MPTCP
        valkeySetError(c, VALKEY_ERR_PROTOCOL, "MPTCP is not supported on this platform");
        return c;
#endif
        c->flags |= VALKEY_MPTCP;
    }
```

Reply:
This file doesn't have IPPROTO_MPTCP because it doesn't `#include` the header where it is defined. Maybe this is not the right place to do the feature check...

When the async API is used, the connect and creation of the context are separated. Connection is done like this:

1. `valkeyAsyncConnectWithOptions()`
2. Attach event loop to async context
3. Run even loop

The context is created in step 1 but the actual connection happens in step 3. I think it's better if we can get the error in step 1. What do you think?

Reply:
I believe the call to connect is done in step 1, as a non-blocking connect. Then in step 3 the socket events will be handled, i.e the client will see that the connect happens.

Reply:
Ah, so in that case, it's OK to do the feature check in `valkeyConnectWithOptions()`.

Reply:
Is this true also for the cluster counterpart?